__tests__/classes/DOMMatrix.js

```diff
@@ -160,4 +145,158 @@ describe('DOMMatrix class', () => {
+      const matrix = new DOMMatrix([4, 5, 1, 3, 10, 9]);
+      const expectedMatrix = new DOMMatrix([
+        4, 5, 0, 0, 1, 3, 0, 0, 0, 0, 1, 0, 610, 1109, 0, 1,
+      ]);
+      const translatedMatrix = matrix.translate(x, y);
+      expect(translatedMatrix.toFloat32Array()).toEqual(
+        expectedMatrix.toFloat32Array()
+      );
+      expect(translatedMatrix.is2D).toEqual(true);
+    });
+
+    it(`should apply 3d changes`, function () {
+      const x = 100;
+      const y = 200;
+      const z = 300;
+      const matrix = new DOMMatrix();
+      const expectedMatrix = new DOMMatrix([
+        1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 100, 200, 300, 1,
+      ]);
+      const translatedMatrix = matrix.translate(x, y, z);
+      expect(translatedMatrix).toEqual(expectedMatrix);
+      expect(translatedMatrix.is2D).toEqual(false);
+    });
+  });
+
+  describe(`scale`, function () {
+    it(`should return a new DOMMatrix instance`, function () {
+      const matrix = new DOMMatrix();
+      const scaledMatrix = matrix.scale(0.5, 0.7);
+      expect(scaledMatrix instanceof DOMMatrix).toBeTruthy();
+      expect(scaledMatrix === matrix).toBeFalsy();
+    });
+
+    it(`should apply 2d changes`, function () {
+      const scaleX = 0.75;
+      const scaleY = 0.5;
+      const matrix = new DOMMatrix([7, 8, 9, 20, 4, 7]);
+      const expectedMatrix = new DOMMatrix([5.25, 6, 4.5, 10, 4, 7]);
+      const scaledMatrix = matrix.scale(scaleX, scaleY);
+      expect(scaledMatrix).toEqual(expectedMatrix);
+    });
+
+    it(`should apply 3d changes`, function () {
+      const scaleX = 0.65;
+      const scaleY = 0.55;
+      const scaleZ = 0.9;
+      const matrix = new DOMMatrix();
+      const expectedMatrix = new DOMMatrix([
+        0.65, 0, 0, 0, 0, 0.55, 0, 0, 0, 0, 0.9, 0, 0, 0, 0, 1,
+      ]);
+      const scaledMatrix = matrix.scale(scaleX, scaleY, scaleZ);
+      expect(scaledMatrix).toEqual(expectedMatrix);
+    });
+  });
+
+  describe(`translateSelf`, function () {
+    it(`should return dot product of a 2d matrix multiplication`, function () {
+      const matrix2D = new DOMMatrix([1, 2, 3, 4, 5, 6]);
+      const tx = 2,
+        ty = 3;
+      const expectedMatrix = new DOMMatrix([1, 2, 3, 4, 16, 22]);
+      matrix2D.translateSelf(tx, ty);
+      expect(matrix2D.toFloat32Array()).toEqual(
+        expectedMatrix.toFloat32Array()
+      );
+      expect(matrix2D.is2D).toEqual(true);
+    });
+
+    it(`should return do product of a 3d matrix`, function () {
+      const matrix3D = new DOMMatrix([
+        1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16,
+      ]);
+      const tx = 2,
+        ty = 3,
+        tz = 4;
+      const expectedMatrix = new DOMMatrix([
+        1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 66, 76, 86, 96,
+      ]);
+      matrix3D.translateSelf(tx, ty, tz);
+      expect(matrix3D.toFloat32Array()).toEqual(
+        expectedMatrix.toFloat32Array()
+      );
+      expect(matrix3D.is2D).toEqual(false);
```

Comment:
Nit: Might prefer `toBeFalsy()` here. Everything else seems reasonable and solid.